### ISSUE TITLE
Compile-AppInBCContainer not using Application ID on downloading symbols

### ISSUE THREAD

**Issue by donmikele75** [microsoft/navcontainerhelper#2867]
We use an Azure DevOps pipeline to compile our applications using Docker Containers created by BcContainerHelper. With updating to BC21, our app has a dependency to "Continia OPplus", publisher "Continia Software" (previously "OPplus 365", publisher "gbedv" - but same App ID) - because of other dependencies (that are still under 20.x version) the old dependency to "OPplus 365", publisher "gbedv" is found, added and queried.

**Output of Compile-AppInBCContainer:**
[...]
Downloading symbols: tegos Group_tegossuite.OPplusConnector_3.0.23010101.0.app
Url : http://172.19.94.10:7049/BC/dev/packages?publisher=tegos%20Group&appName=tegossuite.OPplusConnector&versionText=2.4.0.0&tenant=default
Using WebClient
**Adding dependency to OPplus 365 from gbedv GmbH & Co. KG**
Downloading symbols: tegos gmbh_dorr.Core_21.0.3280.0.app
Url : http://172.19.94.10:7049/BC/dev/packages?publisher=tegos%20gmbh&appName=dorr.Core&versionText=21.0.0.0&tenant=default
Using WebClient
**Downloading symbols: Continia Software_Continia OPplus_21.1.0.27766.app**
Url : http://172.19.94.10:7049/BC/dev/packages?publisher=Continia%20Software&appName=Continia%20OPplus&versionText=21.0.0.0&tenant=default
Using WebClient
Adding dependency to Continia Core from Continia Software

[...]

Downloading symbols: gbedv GmbH & Co. KG_OPplus 365_20.0.0.0.app
Url : http://172.19.94.10:7049/BC/dev/packages?publisher=gbedv%20GmbH%20%26%20Co.%20KG&appName=OPplus%20365&versionText=20.0.0.0&tenant=default
Using WebClient
ERROR Ausnahme beim Aufrufen von "DownloadFile" mit 2 Argument(en):  "Der Remoteserver hat einen Fehler zurückgegeben: (404) Nicht gefunden."
Ausnahme beim Aufrufen von "DownloadFile" mit 2 Argument(en):  "Der Remoteserver hat einen Fehler zurückgegeben: (404) 
Nicht gefunden."
Not Found
No published package matches the provided arguments.
In C:\Program Files\WindowsPowerShell\Modules\BcContainerHelper\4.0.10\AppHandling\Compile-AppInNavContainer.ps1:446 
Zeichen:25

We found out, that VSCode queries dependencies by refering to application id:

[...]

[2023-01-24 12:22:03.98] Sending request to http://dgkesvas06.ad.dorr.de:7049/dorr-dev/dev/packages?publisher=tegos Group&appName=tegossuite.OPplusConnector&versionText=2.4.0.0&appId=ea303d01-764e-416b-9d31-73146942c9d8&tenant=default
[2023-01-24 12:22:03.98] Sending request to http://dgkesvas06.ad.dorr.de:7049/dorr-dev/dev/packages?publisher=Continia Software&appName=Continia OPplus&versionText=21.0.0.0&appId=88cf5d4c-8afc-4a98-9cb7-212196c51d74&tenant=default

[...]

[2023-01-24 12:22:23.52] The following dependencies will be queried for propagated dependencies:
System Application by Microsoft (21.2.0.0)
Base Application by Microsoft (21.2.0.0)
OPplus 365 by gbedv GmbH & Co. KG (20.0.0.0)
Application by Microsoft (20.0.0.0)
TE 365 - ECM Adapter by tegos Holding GmbH (20.0.0.0)
Continia Core by Continia Software (7.0.0.0)
tegossuite.Licensing by tegos Group (2.0.0.0)

[...]

[2023-01-24 12:22:23.53] Sending request to http://dgkesvas06.ad.dorr.de:7049/dorr-dev/dev/packages?publisher=gbedv GmbH %26 Co. KG&appName=OPplus 365&versionText=20.0.0.0&appId=88cf5d4c-8afc-4a98-9cb7-212196c51d74&tenant=default

Is it possible to change Compile-AppInBCContainer to the same logic that is used by VSCode?

[...]

[2023-01-24 12:22:49.64] All reference symbols have been downloaded.

**Comment by freddydk** [comment]
Will get a fix for this asap

**Comment by freddydk** [comment]
This should be fixed in the latest pre-release

**Comment by donmikele75** [comment]
Sorry for my late reply - seems to work only for top level dependencies, but indirect dependencies are still processed by name:

```
Processing dependency tegos Holding GmbH_TE 365 for tegossuite - ECM Adapter_20.0.0.0 (20e66124-454f-478f-bce1-03dcc1c82c5b)
Downloading symbols: tegos Holding GmbH_TE 365 for tegossuite - ECM Adapter_20.0.2452.0.app
Url : http://172.19.90.97:7049/BC/dev/packages?appId=20e66124-454f-478f-bce1-03dcc1c82c5b&versionText=20.0.0.0&tenant=default
Using WebClient
Adding dependency to TE 365 - ECM Adapter from tegos Holding GmbH
Processing dependency Microsoft_System Application_21.2.0.0 ()
Downloading symbols: Microsoft_System Application_21.2.49946.49990.app
Url : http://172.19.90.97:7049/BC/dev/packages?publisher=Microsoft&appName=System%20Application&versionText=21.2.0.0&tenant=default
Using WebClient
Processing dependency Microsoft_Base Application_21.2.0.0 ()
Downloading symbols: Microsoft_Base Application_21.2.49946.49990.app
Url : http://172.19.90.97:7049/BC/dev/packages?publisher=Microsoft&appName=Base%20Application&versionText=21.2.0.0&tenant=default
Using WebClient
Processing dependency tegos Group_tegossuite.Licensing_2.0.0.0 ()
Downloading symbols: tegos Group_tegossuite.Licensing_2.0.23012203.0.app
Url : http://172.19.90.97:7049/BC/dev/packages?publisher=tegos%20Group&appName=tegossuite.Licensing&versionText=2.0.0.0&tenant=default
Using WebClient
Adding dependency to CORE by Simova from Simova GmbH
Processing dependency gbedv GmbH & Co. KG_OPplus 365_20.0.0.0 ()
Downloading symbols: gbedv GmbH & Co. KG_OPplus 365_20.0.0.0.app
Url : http://172.19.90.97:7049/BC/dev/packages?publisher=gbedv%20GmbH%20%26%20Co.%20KG&appName=OPplus%20365&versionText=20.0.0.0&tenant=default
Using WebClient
ERROR Ausnahme beim Aufrufen von "DownloadFile" mit 2 Argument(en):  "Der Remoteserver hat einen Fehler zurückgegeben: (404) Nicht gefunden."
Ausnahme beim Aufrufen von "DownloadFile" mit 2 Argument(en):  "Der Remoteserver hat einen Fehler zurückgegeben: (404) 

```

**Comment by freddydk** [comment]
Which containerhelper version is this?

**Comment by freddydk** [comment]
Got it - should be fixed in 4.0.14

**Comment by donmikele75** [comment]
works like a charm now - thank you very much!
